packages/samples/stencil/src/components/sample-component/sample-component.tsx

```diff
@@ -0,0 +1,123 @@
+import {Bindings, initializeBindings} from '@coveo/atomic';
+import {Component, Element, h, State, forceUpdate} from '@stencil/core';
+import {
+  PagerState,
+  Pager,
+  SearchStatusState,
+  buildSearchStatus,
+  buildPager,
+  Unsubscribe,
+} from '@coveo/atomic/headless';
+
+@Component({
+  tag: 'sample-component',
+  styleUrl: 'sample-component.css',
+  shadow: true,
+})
+export class SampleComponent {
+  private bindings?: Bindings;
+  private error?: Error;
+  private pagerController!: Pager;
+
+  private pagerUnsubscribe: Unsubscribe = () => {};
+  private statusUnsubscribe: Unsubscribe = () => {};
+  private i18nUnsubscribe = () => {};
+
+  @Element() private host!: Element;
+
+  @State() private pagerState!: PagerState;
+  @State() private statusState!: SearchStatusState;
+
+  public async connectedCallback() {
+    try {
```

Comment:
This is an automatically generated sample by the Atomic CLI.

It's necessary when building a custom component using Headless and X Framework (here Stencil) to create the controller and register to its state. The try/catch is there because controller can throw errors on creation if the options are invalid (something that's perhaps not handled in other samples)